app_dart/lib/src/request_handlers/scheduler/scheduler_request_subscription.dart

```diff
@@ -108,16 +162,19 @@ class SchedulerRequestSubscription extends SubscriptionHandler {
 
     for (bbv2.BatchResponse_Response batchResponseResponse in response.responses) {
       if (batchResponseResponse.hasScheduleBuild()) {
-        retry.removeWhere((element) => batchResponseResponse.scheduleBuild.builder == element.scheduleBuild.builder);
+        retry.remove(
```

Comment:
Just checking: It's intentional you changed this from "remove all builders with a matching builder, to the first one"? Is it worth doing `.singleWhere` to bail if we misunderstood something? (Could also be an assert, but won't catch things in prod)

Reply:
Fun fact, builder names (`Linux A`) must be unique across **ALL** ci yamls in all repositories (the builders on LUCI are setup this way). So functionally it has to be identical.

I will change it back though - I was originally caching the firstWhere and using it for something else.

Reply:
I actually changed this to the function creating its own list rather than mutating the containing list. Take another look?